client/ayon_blender/plugins/load/load_image_compositor.py

```diff
@@ -0,0 +1,145 @@
+            "loader": str(self.__class__.__name__),
+            "representation": context["representation"]["id"],
+        }
+        lib.imprint(img_comp_node, data)
+
+        return [img_comp_node]
+
+    def exec_remove(self, container: Dict) -> bool:
+        """Remove the image comp node"""
+        img_comp_node = container["node"]
+        image: Optional[bpy.types.Image] = img_comp_node.image
+
+        # Delete the compositor node
+        bpy.context.scene.node_tree.nodes.remove(img_comp_node)
+
+        # Delete the image if it remains unused
+        self.remove_image_if_unused(image)
+
+        return True
+
+    def exec_update(self, container: Dict, context: Dict):
+        """Update the image comp node to new context version."""
+        path = self.filepath_from_context(context)
+        img_comp_node = container["node"]
+
+        old_image: Optional[bpy.types.Image] = img_comp_node.image
+
+        new_image = bpy.data.images.load(path, check_existing=True)
+        img_comp_node.image = new_image
+
+        self.set_source_and_colorspace(context, img_comp_node)
+        self.remove_image_if_unused(old_image)
+
+    def set_source_and_colorspace(
+        self,
+        context: dict,
+        image_comp_node: bpy.types.CompositorNodeImage
+    ):
+        """
+        Set the image source (e.g. SEQUENCE or FILE), set the duration for
+        a sequence and set colorspace if representation has colorspace data.
+        """
+
+        image = image_comp_node.image
+        representation: dict = context["representation"]
+
+        # Set image source
+        source = "FILE"  # Single image file
+        if representation["context"].get("udim"):
+            source = "UDIM"
+        elif representation["context"].get("frame"):
+            source = "SEQUENCE"
+        else:
+            ext = os.path.splitext(image.filepath)[-1]
+            if ext in VIDEO_EXTENSIONS:
+                source = "MOVIE"
+
+        image.source = source
+
+        # Set duration on the compositor node if sequence is used
+        if source in {"SEQUENCE", "MOVIE"}:
+            version_attrib: dict = context["version"]["attrib"]
+            frame_start = version_attrib.get("frameStart", 0)
+            frame_end = version_attrib.get("frameEnd", 0)
+            handle_start = version_attrib.get("handleStart", 0)
+            handle_end = version_attrib.get("handleEnd", 0)
+            frame_start_handle = frame_start - handle_start
+            frame_end_handle = frame_end + handle_end
+            duration: int = frame_end_handle - frame_start_handle + 1
+            image_comp_node.frame_duration = duration
+            if source == "SEQUENCE":
+                image_comp_node.frame_start = frame_start_handle
+                image_comp_node.frame_offset = frame_start_handle - 1
+            else:
+                image_comp_node.frame_start = frame_start_handle
+                image_comp_node.frame_offset = 0
```

Comment:
Honestly. These were just magical... 

For whatever reason a file sequence with frame start 1001 defaults playback at frame 0, whereas a video at frame start 1001 defaults the playback at frame 1001. Hence, the sequence one requires an offset - and because it does not start at frame 1 but zero it needs to be minus one. 🤷 

It must be that I'm stupid - but if any blender geek can tell me how this _should_ be approach in the compositor to configure a video or sequence to start at the actual frame I want without this that'd be more than welcome.